src/main/scala-2.12/com/linagora/gatling/imap/protocol/command/CompressCommand.scala

```diff
@@ -0,0 +1,27 @@
+package com.linagora.gatling.imap.protocol.command
+
+import java.nio.charset.StandardCharsets
+
+import com.yahoo.imapnio.async.request.{ImapCommandType, ImapRequestAdapter}
+import io.netty.buffer.{ByteBuf, Unpooled}
+
+case object CompressCommandType extends ImapCommandType {
+  override def getType: String = "COMPRESS"
+}
+
+class CompressCommand(algorithm: String) extends ImapRequestAdapter {
```

Comment:
That's not enough as the client side also need to set up compression.

Actually the ImapNIO session have a dedicated method for compression, that do not rely on commands.